### PR TITLE
[KernelGen][Nvidia] Add special_i0 operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a Triton kernel for `special_i0` operator. Computes the modified Bessel function of the first kind of order 0.

## Testing
- Parametrized tests over multiple shapes `[(2, 3), (128, 256), (512, 512)]` and `FLOAT_DTYPES`
- Validated against PyTorch reference on device side
- Tested on: Nvidia, Tianshu, Muxi, Ascend, Hygon

## Performance
Test command: `pytest benchmark/test_special_i0.py --level core` (NVIDIA H20)

| Configuration | Torch Latency (ms) | Gems Latency (ms) | Speedup | TFLOPS |
|---|---|---|---|---|
| 1073741824 | 5.434 | 2.904 | 1.876 | 0.371 |
| 64, 64 | 0.007 | 0.006 | 1.234 | 0.001 |
| 4096, 4096 | 0.091 | 0.051 | 1.780 | 0.329 |
| 64, 512, 512 | 0.091 | 0.051 | 1.772 | 0.327 |
| 1024, 1024, 1024 | 5.442 | 2.913 | 1.872 | 0.370 |
| **Arithmetic Mean** | — | — | **1.707** | — |

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (30 cases) | 1.707 | Primary |
| Tianshu | PASS | PASS (42 cases) | 1.625 | — |
| Muxi | PASS | PASS (42 cases) | 1.003 | — |
| Ascend | PASS | N/A | — | — |
| Hygon | PASS | PASS (42 cases) | 1.108 | — |

## Files Changed
- `src/flag_gems/ops/special_i0.py`: Triton kernel implementation
- `tests/test_special_i0.py`: Accuracy test
- `benchmark/test_special_i0.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Math, stage: alpha 5.1)

### CODE REVIEW

File: tests/test_special_i0.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        ref_out = torch.ops.aten.special_i0(ref_x.float()).to(dtype)
+    else:
+        ref_out = torch.ops.aten.special_i0(ref_x)
+    with flag_gems.use_gems():
+        act_out = torch.ops.aten.special_i0(x)
+    utils.gems_assert_close(act_out, ref_out, dtype)
+
+
+@pytest.mark.special_i0
+@pytest.mark.parametrize("shape", [(2, 3), (128, 256), (512, 512)])
```
Comment (lukatao) [inline tests/test_special_i0.py]: use shapes from utils

File: benchmark/test_special_i0.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@pytest.mark.special_i0
+def test_special_i0_out():
+    def torch_op(x):
+        out = torch.empty_like(x)
+        torch.ops.aten.special_i0.out(x, out=out)
+        return out
+
+    bench = base.UnaryPointwiseBenchmark(
+        op_name="special_i0",
+        torch_op=torch_op,
```
Comment (w1120029931-bit) [inline benchmark/test_special_i0.py]: Do not encapsulate this function

File: tests/test_special_i0.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    if dtype in (torch.float16, torch.bfloat16):
+        ref_out = torch.ops.aten.special_i0(ref_x.float()).to(dtype)
+    else:
+        ref_out = torch.ops.aten.special_i0(ref_x)
+    with flag_gems.use_gems():
+        act_out = torch.ops.aten.special_i0(x)
+    utils.gems_assert_close(act_out, ref_out, dtype)
+
+
+@pytest.mark.special_i0
```
Comment (w1120029931-bit) [inline tests/test_special_i0.py]: should be mark.special_i0_out?

File: conf/operators.yaml [hunk 4]
```diff
@@ -5949,6 +5949,17 @@ ops:
       - DSA
     stages:
       - beta: '5.0'
+  - id: special_i0
```
Comment (w1120029931-bit) [inline conf/operators.yaml:5952]: Missing id special_i0_out?

Comment (bwbwzzz) [reply conf/operators.yaml:5952]: Thanks for the pointers! I’ve made the changes. Could you give it another review ?

File: tests/test_special_i0.py (hunk 3 above)
Comment (bwbwzzz) [reply tests/test_special_i0.py]: yes，thanks，have revised

File: benchmark/test_special_i0.py (hunk 2 above)
Comment (bwbwzzz) [reply benchmark/test_special_i0.py]: thanks, have revised,

**Review by Dongxu-H — CHANGES_REQUESTED** [review]
## 代码审查：special_i0

### 🔴 关键问题

**1. 合并冲突需要解决**
此 PR 当前处于 CONFLICTING 状态，需要 rebase 到最新的 master 分支并解决冲突。

**2. 与 PR #3689 功能重叠**
PR #3689 也实现了 `special_i0`，两个实现的主要区别：
- 本 PR 使用手写的 kernel launch 逻辑
- PR #3689 可能使用不同的辅助函数

建议协调这两个 PR，选择最佳实现。

### 🟡 建议改进

**1. 代码完整性**
diff 显示被截断，但我看到的实现有以下建议：

```python
# BLOCK_SIZE=1024 chosen to balance occupancy and register pressure for H20
```
注释提到 H20，建议说明这是针对哪种 GPU 架构的优化。

**2. 类型转换逻辑**
```python
if not x_in.is_floating_point():
    x_in = x_in.to(torch.get_default_dtype())
```
建议添加警告日志，说明发生了隐式类型转换。

**3. 测试覆盖**
请确保测试包含：
- 边界值（|x| = 3.75）
- 大值输入（|x| > 100）验证数值稳定性
- 不同 dtype 的精度对比

**4. 代码重复**
注意到 `special_i0.py` 和 `special_i0e.py` 可能有相似的多项式系数。建议考虑提取公共代码到共享模块。

### ✅ 优点

- 使用经典的多项式近似方法
- 边界处理清晰（|x| <= 3.75 分段）
- 支持非连续张量

### 📝 建议添加的测试

```python
def test_special_i0_boundary():
    # 测试边界值
    x = torch.tensor([3.75, -3.75], dtype=torch.float32, device=flag_gems.device)
    ...

def test_special_i0_large_values():
    # 测试大值稳定性
    x = torch.tensor([100.0, 500.0], dtype=torch.float32, device=flag_gems.device)
    ...
```

---
请解决合并冲突，并与 PR #3689 协调后重新提交。

File: src/flag_gems/ops/special_i0.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+@triton.jit
+def _special_i0_kernel(x_ptr, out_ptr, n_elements, BLOCK_SIZE: tl.constexpr):
+    pid = tl.program_id(axis=0)
+    block_start = pid * BLOCK_SIZE
+    offsets = block_start + tl.arange(0, BLOCK_SIZE)
+    mask = offsets < n_elements
+
+    x = tl.load(x_ptr + offsets, mask=mask, other=0.0)
+    x_f32 = x.to(tl.float32)
+    ax = tl.abs(x_f32)
```
Comment (bwbwzzz) [inline src/flag_gems/ops/special_i0.py:20]: The calculation is hardcoded to FP32, which may cause precision loss.
